libderections-common/src/main/java/com/mapbox/navigation/route/common/NavigationRoute.kt

```diff
@@ -0,0 +1,782 @@
+package com.mapbox.navigation.route.common
+
+import android.content.Context
+import androidx.annotation.FloatRange
+import androidx.annotation.IntRange
+import com.mapbox.api.directions.v5.DirectionsCriteria
+import com.mapbox.api.directions.v5.MapboxDirections
+import com.mapbox.api.directions.v5.models.DirectionsResponse
+import com.mapbox.api.directions.v5.models.RouteOptions
+import com.mapbox.core.exceptions.ServicesException
+import com.mapbox.core.utils.TextUtils
+import com.mapbox.geojson.Point
+import com.mapbox.navigation.route.common.extension.getUnitTypeForLocale
+import com.mapbox.navigation.utils.extensions.inferDeviceLocale
+import okhttp3.EventListener
+import okhttp3.Interceptor
+import retrofit2.Call
+import retrofit2.Callback
+
+/**
+ * The NavigationRoute class wraps the [MapboxDirections] class with parameters which
+ * <u>must</u> be set in order for a navigation session to successfully begin. While it is possible
+ * to pass in any [com.mapbox.api.directions.v5.models.DirectionsRoute] into
+ * [MapboxNavigation.startNavigation], using this class will ensure your
+ * request includes all the proper information needed for the navigation session to begin.
+ *
+ * Developer Note: MapboxDirections cannot be directly extended since it is an AutoValue class.
+ *
+ * 0.5.0
+ */
+
+class NavigationRoute
+internal constructor(
+    private val mapboxDirections: MapboxDirections
+) {
+
+    companion object {
+        private val EVENT_LISTENER = NavigationRouteEventListener()
+
+        /**
+         * Build a new [NavigationRoute] object with the proper navigation parameters already setup.
+         *
+         * @return a [Builder] object for creating this object
+         * @since 0.5.0
+         */
+        @JvmStatic
+        fun builder(context: Context): Builder =
+            Builder()
+                .annotations(
+                    DirectionsCriteria.ANNOTATION_CONGESTION,
+                    DirectionsCriteria.ANNOTATION_DISTANCE
+                )
+                .language(context)
+                .voiceUnits(context)
+                .profile(DirectionsCriteria.PROFILE_DRIVING_TRAFFIC)
+                .continueStraight(true)
+    }
+
+    /**
+     * Wrapper method for Retrofit's [Call.clone] call, useful for getting call information
+     * and allowing you to perform additional functions on this [NavigationRoute] class.
+     *
+     * @return cloned call
+     * @since 1.0.0
+     */
+    val call: Call<DirectionsResponse>
+        get() = mapboxDirections.cloneCall()
+
+    /**
+     * Call when you have constructed your navigation route with your desired parameters. A
+     * [Callback] must be passed into the method to handle both the response and failure.
+     *
+     * @param callback a RetroFit callback which contains an onResponse and onFailure
+     * @since 0.5.0
+     */
+    fun getRoute(callback: Callback<DirectionsResponse>) {
```

Comment:
i'd discourage leaking retrofit via public signatures. unless `libdirections-common` is used only by internal implementations.

Reply:
it's internal only. Used in `common` module